compiler/src/iree/compiler/InputConversion/Common/test/tensor_ops_to_linalg.mlir

```diff
@@ -0,0 +1,297 @@
+// RUN: iree-opt --pass-pipeline="builtin.module(func.func(iree-tensor-ops-to-linalg))" --split-input-file %s | FileCheck %s
+
+// first all tests where i assume no reduced dims
+func.func @basic(%arg0: tensor<2x1xi64>, %arg1: tensor<16x4xf32>) -> tensor<2x1x4xf32> {
+  %gather = tensor.gather %arg1[%arg0] gather_dims([0]) : (tensor<16x4xf32>, tensor<2x1xi64>) -> tensor<2x1x4xf32>
+  return %gather : tensor<2x1x4xf32>
+}
+
+// CHECK: #[[MAP0:.*]] = affine_map<(d0, d1, d2) -> (d0, d1, d2)>
+// CHECK: func.func @basic
+// CHECK-NOT: tensor.gather
+// CHECK: %[[C0:.*]] = arith.constant 0
+// CHECK: tensor.empty
+// CHECK: %[[RES:.+]] = linalg.generic
+// CHECK-SAME: indexing_maps = [#[[MAP0]]]
+// CHECK: %[[LINALG_IDX0:.*]] = linalg.index 0
+// CHECK: %[[LINALG_IDX2:.*]] = linalg.index 2
+// CHECK: %[[INDICES_IDX0:.*]] = tensor.extract %arg0[%[[LINALG_IDX0]], %[[C0]]]
```

Comment:
This looks off, `%arg0` should just be the operand to `linalg.generic`.
I see now why you had the constants in the range of the indexing maps. There are two options here
1) Keep the `0` in the indexing maps. Subsequent transformations in IREE will clean those up. 
2) Or add a reshape to `%arg0` that drops the extra-unit dim so that you can get the indexing map to not contain the `0`. That is essentially what the transformations in IREE do. 

I know I said (1) is not a good idea, but I think on looking at this, I can live with that.